pages/workspaces/[workspaceId]/settings.tsx

```diff
@@ -79,87 +116,149 @@ const WorkspaceSettings = ({ workspace }: WorkspaceSettingsProps) => {
-            description,
-            sessionToken: sessionToken!,
-          });
-
-          if (error) {
-            toast({ description: `Workspace update failed`, variant: "danger" });
-          } else {
-            setWorkspaceName(name);
-            data && setWorkspaces([...workspaces.filter((w) => w.id !== workspace.id), data]);
-            toast({ description: `Workspace updated successfully`, variant: "success" });
-          }
-        }}
-      >
-        <TextInput
-          name="name"
-          label="Workspace Name"
-          defaultValue={workspace.name}
-          placeholder="Workspace name"
-          className="w-full md:w-max"
-          required
-        />
-        <TextInput
-          name="description"
-          label="Workspace Description"
-          defaultValue={workspace.description}
-          placeholder="Workspace description"
-          className="w-full md:w-3/4 max-w-lg"
-        />
-        <div className="bg-white sticky-bottom fixed bottom-0 right-0 self-end m-6">
-          <Button variant="primary" className="flex gap-2.5 items-center cursor-pointer w-min mt-2 sm:mt-0 self-end">
-            Update Workspace
-          </Button>
+      <div className="grid gap-6">
+        <div>
+          <h1 className="border-b bottom pb-4">Workspace Settings</h1>
+          <form className="flex flex-col pt-6 gap-6" onSubmit={updateWorkspace}>
+            <TextInput
+              name="name"
+              label="Workspace Name"
+              defaultValue={workspace.name}
+              placeholder="Workspace name"
+              className="w-full md:w-max"
+              required
+            />
+            <TextInput
+              name="description"
+              label="Workspace Description"
+              defaultValue={workspace.description}
+              placeholder="Workspace description"
+              className="w-full md:w-3/4 max-w-lg"
+            />
+            <div className="bg-white sticky-bottom fixed bottom-0 right-0 self-end m-6">
+              <Button
+                variant="primary"
+                className="flex gap-2.5 items-center cursor-pointer w-min mt-2 sm:mt-0 self-end"
+              >
+                Update Workspace
+              </Button>
+            </div>
+          </form>
         </div>
-      </form>
-      <div className="flex flex-col gap-4 py-6">
-        <Title className="!text-1xl !leading-none py-6" level={4}>
-          Danger Zone
-        </Title>
-
-        <div className="flex flex-col p-6 rounded-2xl bg-light-slate-4">
-          <Title className="!text-1xl !leading-none !border-light-slate-8 border-b pb-4" level={4}>
-            Delete Workspace
+        <TrackedReposTable
+          repositories={trackedRepos}
+          onAddRepos={() => {
+            setTrackedReposModalOpen(true);
+          }}
+          onRemoveTrackedRepo={(event) => {
+            const { repo } = event.currentTarget.dataset;
+
+            if (!repo) {
```

Comment:
We should never end up in this `if` block. This could be removed and switch

```typescript
const { repo } = event.currentTarget.dataset;
```

to

```typescript
const repo = event.currentTarget.dataset.repo!;
```